volatility3/framework/plugins/linux/psscan.py

```diff
@@ -0,0 +1,127 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+from typing import Iterable, List, Tuple
+import struct
+
+from volatility3.framework import renderers, interfaces, symbols, constants
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.layers import scanners
+from volatility3.framework.renderers import format_hints
+
+
+class PsScan(interfaces.plugins.PluginInterface):
+    """Scans for processes present in a particular linux image."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    def _get_task_fields(
+        self, task: interfaces.objects.ObjectInterface
+    ) -> Tuple[int, int, int, str]:
+        """Extract the fields needed for the final output
+
+        Args:
+            task: A task object from where to get the fields.
+        Returns:
+            A tuple with the fields to show in the plugin output.
+        """
+        pid = task.tgid
+        tid = task.pid
+        ppid = task.parent.tgid if task.parent else 0
+        name = utility.array_to_string(task.comm)
+
+        task_fields = (format_hints.Hex(task.vol.offset), pid, tid, ppid, name)
+        return task_fields
+
+    def _generator(self):
+        """Generates the tasks found from scanning."""
+
+        for task in self.scan_tasks(
+            self.context, self.config["kernel"], self.config["kernel.symbol_table_name"]
```

Comment:
This is... kinda okay, but if the key isn't present or the kernel module is generated in some other way, then `kernel.symbol_table_name` might not be what we're after.  I think the correct method for getting this value is to get the kernel from the list, and then ask it from which symbol table name it has.  Sorry, I know it's a very corner case, but I'm not sure we can strictly rely on that key always existing, only the direct requirements on this plugin or attributes of objects that fulfill the direct requirements of this plugin.

Reply:
I think I've now done this in a better way - what do you think?